challenge_opener/src/main/java/com/example/challengeopener/ChallengesAdapter.kt

```diff
@@ -0,0 +1,59 @@
+package com.example.challengeopener
+
+import android.content.Context
+import android.content.Intent
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.content.res.AppCompatResources.getDrawable
+import androidx.core.content.ContextCompat.startActivity
+import androidx.recyclerview.widget.RecyclerView
+import com.example.challengeopener.databinding.RecyclerviewChallengesItemBinding
+
+class ChallengesAdapter(private val context: Context, private var list: List<NewActivityInfo>) :
+    RecyclerView.Adapter<ChallengesAdapter.ChallengesHolder>() {
+
+    inner class ChallengesHolder(private var binding: RecyclerviewChallengesItemBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+        fun bind(position: Int) {
+            val item = list[position]
+            binding.run {
+                constraintlayoutChallengesDetailContainer.visibility = View.GONE
+                txtviewChallengesDetailDescription.text = item.activityDescription
+                btnChallengesOpener.text = item.activityTitle
+                btnChallengesOpener.icon = getDrawable(context, item.activityIcon)
```

Comment:
@js-gonzalez You could either use the context from item view or pass a custom class that provides your resources.

```suggestion
                btnChallengesOpener.icon = getDrawable(itemView.context, item.activityIcon)
```

or

```suggestion
                btnChallengesOpener.icon = resourceProvider.getIcon(item.activityIcon)
```